FlowCrypt/src/main/java/com/flowcrypt/email/database/dao/MessageDao.kt

```diff
@@ -506,6 +506,25 @@ abstract class MessageDao : BaseDao<MessageEntity> {
     }
   }
 
+  open suspend fun updateGmailLabels(
+    email: String?,
+    label: String?,
+    labelsToBeUpdatedMap: Map<Long, String>
+  ) =
+    withContext(Dispatchers.IO) {
+      if (email == null || label == null) {
+        return@withContext
+      }
+
+      val messagesToBeUpdated = mutableListOf<MessageEntity>()
+      val entities = getMsgsByUIDs(email, label, labelsToBeUpdatedMap.keys.toList())
+      for (entity in entities) {
+        messagesToBeUpdated.add(entity.copy(labelIds = labelsToBeUpdatedMap[entity.uid]))
+      }
```

Comment:
Better to add map for this case. We don't need to declear above messagesToBeUpdated in this case.

```kt
val updatedMessages = entities.map { entity ->
  entity.copy(labelIds = labelsToUpdate[entity.uid])
}
```